hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -555,6 +555,10 @@ public void refreshTimeline() throws IOException {
       case INSERT_OVERWRITE_TABLE:
         writeStatusRDD = writeClient.insertOverwriteTable(records, instantTime).getWriteStatuses();
         break;
+      case DELETE_PARTITION:
+        List<String> partitions = records.map(record -> record.getPartitionPath()).distinct().collect();
```

Comment:
The logic here takes the partition list from the input records and does nothing to the records.  This workflow looks strange to me.  Shouldn't the input be the partition list directly?  It's counter-intuitive for the users to provide a list of records for deleting corresponding partitions.

Reply:
As discussed offline, `DELETE_PARTITION` should be added as a separate operation here, as to how this PR adds the functionality.  To leverage this new op, user needs to pull some data from the existing partitions which are intended to be deleted, for the Deltastreamer to ingest.  We'll keep this logic for now.